### ISSUE TITLE
No error emitted when declaring RayQuery object in constant buffer

### ISSUE THREAD

**Issue by tex3d** [microsoft/DirectXShaderCompiler#7203]
**Description**
The compiler should emit an error diagnostic if you declare a RayQuery in a cbuffer, since it makes no sense to have a RayQuery in a constant buffer.

**Steps to Reproduce**
https://godbolt.org/z/bvEK7bWvT

**Actual Behavior**
The compiler actually puts the object with the fake handle in the constant buffer that looks like:
```
; cbuffer $Globals
; {
;
;   struct $Globals
;   {
;
;       struct class.RayQuery<0>
;       {
;
;           uint h;                                   ; Offset:    0
;       
;       } rq;                                         ; Offset:    0
;
;       uint j;                                       ; Offset:    4
;   
;   } $Globals;                                       ; Offset:    0 Size:     8
;
; }
```

If you try to use the RayQuery object, the compiler will crash.

**Environment**
- DXC version: latest main
- Host Operating System: any
